#55985 - Remove Google fonts from Twenty Seventeen

Luehrsen opened the pull request:
This pull request aims to remove google webfont requests from core themes and serve needed fonts locally.

Trac ticket: https://core.trac.wordpress.org/ticket/55985#ticket

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

Luehrsen committed:
Register 'Libre Franklin' in its own stylesheet

Luehrsen committed:
Check in the additional font files

Luehrsen committed:
Add package-lock for whatever reason

Luehrsen committed:
Register the correct path for the fonts

Luehrsen committed:
Return our local file if requested by the font

Luehrsen committed:
Fix indentation

Luehrsen committed:
Revert "Add package-lock for whatever reason"

This reverts commit bf8871fae6f92c0e7bec5a956be901b802ae9603.

j-hoffmann commented:
The new `@font-face` rules use `local('')` in the `src:` declarations. 

On the github repository for the [Google Webfonts Helper there is an open issue](https://github.com/majodev/google-webfonts-helper/issues/84) about problems arising from using such declarations. 
[There is another open issue](https://github.com/majodev/google-webfonts-helper/issues/124) about the same type of declaration where I commented having [another type of problem](https://github.com/majodev/google-webfonts-helper/issues/124#issuecomment-896124234) myself. I know I should have opened a separate ticket for my issue, but I realized that the project didn't seem to be mantained anymore (last code change is 5 years old), so I disited.

To avoid these problems, I would suggest to either use the "smiley" hack as mentioned [here](https://stackoverflow.com/questions/3698319/css-font-face-what-does-src-local-mean) or drop the `local('')` part altogether.

Luehrsen commented:
> To avoid these problems, I would suggest to either use the "smiley" hack as mentioned [here](https://stackoverflow.com/questions/3698319/css-font-face-what-does-src-local-mean) or drop the `local('')` part altogether.

I think dropping local is the way to go.

Luehrsen committed:
Remove the local parameter

jffng reviewed (commented):
I think fonts typically come with a license that should be bundled as well, for example https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-content/themes/twentytwentytwo/assets/fonts/ibm-plex/LICENSE.txt

cbirdsong commented:
I see this PR includes the font files in the repo, but installing them via NPM is also an option: https://fontsource.org/

Luehrsen commented:
> I see this PR includes the font files in the repo, but installing them via NPM is also an option: https://fontsource.org/

This is how I would do it today. But I think the overhead in terms of packages and build tools is a little too much for the scope of what we want to achieve here. I don't think introducing a new package and build process for a legacy polyfill is wise here. 

We can always go back and optimise if we wish to do that.

Luehrsen committed:
Add the license text

2ndkauboy commented:
> I think dropping local is the way to go.

Why not using the real local name `local('Libre Franklin')`? On my Manjaro laptop I have the Google Fonts package installed and load pretty much any Goole Web Font locally. Some operating systems have other local fonts installed by default.

j-hoffmann commented:
> Why not using the real local name `local('Libre Franklin')`? On my Manjaro laptop I have the Google Fonts package installed and load pretty much any Goole Web Font locally. Some operating systems have other local fonts installed by default.

Apparently there were cases with colliding font names. Originally Google Fonts used to add the `local("Font Name")` delarations but eventually [removed them except for specific cases](https://twitter.com/googlefonts/status/1323363090969333762).

Another possible reason is that there is no control over which version of the font is loaded from the user's computer. The user could have a version with a limited character set or even incomplete variations installed.

Luehrsen committed:
Move the fonts into the assets folder

Luehrsen committed:
Implement modern approach with unicode-range

Luehrsen commented:
Based on a discussion with @2ndkauboy I've rewritten the implementation and added the unicode-range selector. (https://github.com/WordPress/wordpress-develop/pull/2823#discussion_r900925587)

The downside is, that the font.css is really complex now and we have a lot of files. But the upside is, that character subsets get automatically loaded when needed. 

https://user-images.githubusercontent.com/1105871/174487580-8919f358-1bb4-49bd-a02e-2c3ac01e8e2d.mov

2ndkauboy commented:
> I've rewritten the implementation and added the unicode-range selector.

That seems like a great solution!

2ndkauboy commented:
I believe locally hosted fonts are the best way forward, but this is really interesting:

https://twitter.com/dimensionmedia/status/1538912055381463041

Luehrsen commented:
IMHO not without asking for consent first.

> Am 21.06.2022 um 16:01 schrieb Bernhard Kau ***@***.***>:
> 
> ﻿
> I believe locally hosted fonts are the best way forward, but this is really interesting:
> 
> https://twitter.com/dimensionmedia/status/1538912055381463041
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.

2ndkauboy commented:
> IMHO not without asking for consent first.

I think in terms of GDPR compliance it is OK without consent, as no personal data is sent to any non-EU entity.

But still I think for the default themes we should really go with local fonts.

Luehrsen committed:
Merge remote-tracking branch 'upstream/trunk' into update/google-fonts-legacy-themes

Luehrsen committed:
Remove the preconnect, as that leaks IP to google (GDPR)
